main.go

```diff
@@ -0,0 +1,346 @@
+	"github.com/openshift/managed-upgrade-operator/controllers/nodekeeper"
+	"github.com/openshift/managed-upgrade-operator/controllers/upgradeconfig"
+	cv "github.com/openshift/managed-upgrade-operator/pkg/clusterversion"
+	"github.com/openshift/managed-upgrade-operator/pkg/collector"
+	"github.com/openshift/managed-upgrade-operator/pkg/configmanager"
+	"github.com/openshift/managed-upgrade-operator/pkg/drain"
+	"github.com/openshift/managed-upgrade-operator/pkg/eventmanager"
+	"github.com/openshift/managed-upgrade-operator/pkg/k8sutil"
+	"github.com/openshift/managed-upgrade-operator/pkg/machinery"
+	"github.com/openshift/managed-upgrade-operator/pkg/metrics"
+	"github.com/openshift/managed-upgrade-operator/pkg/scheduler"
+	"github.com/openshift/managed-upgrade-operator/pkg/upgradeconfigmanager"
+	ucm "github.com/openshift/managed-upgrade-operator/pkg/upgradeconfigmanager"
+	cub "github.com/openshift/managed-upgrade-operator/pkg/upgraders"
+	"github.com/openshift/managed-upgrade-operator/pkg/validation"
+	"github.com/openshift/managed-upgrade-operator/util"
+	"github.com/openshift/managed-upgrade-operator/version"
+
+	opmetrics "github.com/openshift/operator-custom-metrics/pkg/metrics"
+
+	"github.com/operator-framework/operator-lib/leader"
+
+	configv1 "github.com/openshift/api/config/v1"
+	machineapi "github.com/openshift/api/machine/v1beta1"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monclientv1 "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned/typed/monitoring/v1"
+
+	routev1 "github.com/openshift/api/route/v1"
+
+	machineconfigapi "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	metricsHost             = "0.0.0.0"
+	metricsPort       int32 = 8383
+	customMetricsPath       = "/metrics"
+	scheme                  = apiruntime.NewScheme()
+	setupLog                = ctrl.Log.WithName("setup")
+)
+var log = logf.Log.WithName("cmd")
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(machineconfigapi.AddToScheme(scheme))
+	utilruntime.Must(upgradev1alpha1.AddToScheme(scheme))
+	utilruntime.Must(monitoringv1.AddToScheme(scheme))
+	utilruntime.Must(routev1.Install(scheme))
+	utilruntime.Must(configv1.Install(scheme))
+	utilruntime.Must(machineapi.Install(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func printVersion() {
+	log.Info(fmt.Sprintf("Operator Version: %s", version.Version))
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+	log.Info(fmt.Sprintf("Version of operator-sdk: v%v", version.SDKVersion))
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
```

Comment:
A minor thing, but this variable is never used anywhere, the `metricsPort` variable is the one that is actually used for setting up the metrics in the code. 

Perhaps it should be changed to `:8383` as a default, and the global `metricsHost` / `metricsPort` are removed, and metricsAddr` is used in the manager setup.

Reply:
Thanks Matt! I have changed this. However, the `metricsPort` is used in `addMetrics()` function to create the service too so I have maintained the `metricsPort` var and used it commonly to initialise the `metricsAddr` var too. Let me know if this seems ok to you.